db/migrate/20191213123801_create_whitehall_imported_assets.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class CreateWhitehallImportedAssets < ActiveRecord::Migration[6.0]
+  def change
+    create_table :whitehall_imported_assets do |t|
+      t.references :whitehall_import,
+                   foreign_key: { to_table: :whitehall_imports,
+                                  on_delete: :restrict },
+                   index: true,
+                   null: false
+
+      # Asset must be one of FileAttachment or Image
+      # @TODO - write a custom validator?
+      t.references :file_attachment_revision,
+                   foreign_key: { to_table: :file_attachment_revisions,
+                                  on_delete: :cascade }
+      t.references :image_revision,
+                   foreign_key: { to_table: :image_revisions,
+                                  on_delete: :cascade }
```

Comment:
Both these references should also be restricts as well for ADR consistency.

Also at a purely semantic level I don't think we'd want this record deleted automatically as it'd leave the asset in the abyss.